Mathlib/CategoryTheory/ChosenFiniteProducts.lean

```diff
@@ -195,6 +195,263 @@ instance (priority := 100) : Limits.HasFiniteProducts C :=
+@[reassoc]
+theorem prodComparison_natural_whiskerRight (f : A ⟶ A') :
+    F.map (f ▷ B) ≫ prodComparison F A' B =
+      prodComparison F A B ≫ (F.map f ▷ F.obj B) := by
+  rw [← tensorHom_id, prodComparison_natural, Functor.map_id]
+  rfl
+
+section
+variable [IsIso (prodComparison F A B)]
+
+/-- If the product comparison morphism is an iso, its inverse is natural in both argument. -/
+@[reassoc]
+theorem prodComparison_inv_natural (f : A ⟶ A') (g : B ⟶ B') [IsIso (prodComparison F A' B')] :
+    inv (prodComparison F A B) ≫ F.map (f ⊗ g) =
+      (F.map f ⊗ F.map g) ≫ inv (prodComparison F A' B') := by
+  rw [IsIso.eq_comp_inv, Category.assoc, IsIso.inv_comp_eq, prodComparison_natural]
+
+/-- If the product comparison morphism is an iso, its inverse is natural in the right argument. -/
+@[reassoc]
+theorem prodComparison_inv_natural_whiskerLeft (g : B ⟶ B') [IsIso (prodComparison F A B')] :
+    inv (prodComparison F A B) ≫ F.map (A ◁ g) =
+      (F.obj A ◁ F.map g) ≫ inv (prodComparison F A B') := by
+  rw [IsIso.eq_comp_inv, Category.assoc, IsIso.inv_comp_eq, prodComparison_natural_whiskerLeft]
+
+/-- If the product comparison morphism is an iso, its inverse is natural in the left argument. -/
+@[reassoc]
+theorem prodComparison_inv_natural_whiskerRight (f : A ⟶ A') [IsIso (prodComparison F A' B)] :
+    inv (prodComparison F A B) ≫ F.map (f ▷ B) =
+      (F.map f ▷ F.obj B) ≫ inv (prodComparison F A' B) := by
+  rw [IsIso.eq_comp_inv, Category.assoc, IsIso.inv_comp_eq, prodComparison_natural_whiskerRight]
+
+end
+
+/-- The product comparison morphism from `F(A ⊗ -)` to `FA ⊗ F-`, whose components are given by
+`prodComparison`. -/
+@[simps]
+def prodComparisonNatTrans (A : C) :
+    (curriedTensor C).obj A ⋙ F ⟶ F ⋙ (curriedTensor D).obj (F.obj A) where
+  app B := prodComparison F A B
+  naturality x y f := by
+    apply hom_ext <;>
+    simp only [Functor.comp_obj, curriedTensor_obj_obj,
+      Functor.comp_map, curriedTensor_obj_map, Category.assoc, prodComparison_fst, whiskerLeft_fst,
+      prodComparison_snd, prodComparison_snd_assoc, whiskerLeft_snd, ← F.map_comp]
+
+/-- The product comparison morphism from `F(- ⊗ -)` to `F- ⊗ F-`, whose components are given by
+`prodComparison`. -/
+@[simps]
+def prodComparisonBifunctorNatTrans :
+    curriedTensor C ⋙ (whiskeringRight _ _ _).obj F ⟶
+      F ⋙ curriedTensor D ⋙ (whiskeringLeft _ _ _).obj F where
+  app A := prodComparisonNatTrans F A
+  naturality x y f := by
+    ext z
+    apply hom_ext <;>
+    simp only [Functor.comp_obj, whiskeringRight_obj_obj, curriedTensor_obj_obj,
+      whiskeringLeft_obj_obj, Functor.comp_map, whiskeringRight_obj_map, NatTrans.comp_app,
+      whiskerRight_app, curriedTensor_map_app, prodComparisonNatTrans_app, Category.assoc,
+      prodComparison_fst, whiskeringLeft_obj_map, whiskerLeft_app, whiskerRight_fst,
+      prodComparison_fst_assoc, prodComparison_snd, whiskerRight_snd, ← F.map_comp]
```

Comment:
As this proof is very fast, I would suggest using this syntax:
```suggestion
    apply hom_ext <;> simp [← Functor.map_comp]
```
Same for `prodComparisonNatTrans`.